compiler/src/iree/compiler/GlobalOptimization/DemoteContractionInputsToBF16.cpp

```diff
@@ -65,32 +68,34 @@ struct DemoteContractionInputsToBF16Pattern
     }
 
     auto replaceOpInputs = [&](auto *typePtr) {
-      auto namedOp = cast<std::remove_pointer_t<decltype(typePtr)>>(op);
+      auto namedOp = cast<std::remove_pointer_t<decltype(typePtr)>>(linalgOp);
       rewriter.replaceOpWithNewOp<std::remove_pointer_t<decltype(typePtr)>>(
           linalgOp, demotedInputs, linalgOp.getDpsInits(),
           linalg::getPrunedAttributeList(namedOp));
     };
 
-    if (isa<linalg::MatmulOp>(op)) {
+    if (isa<linalg::MatmulOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::MatmulOp *>(nullptr));
-    } else if (isa<linalg::MatvecOp>(op)) {
+    } else if (isa<linalg::MatvecOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::MatvecOp *>(nullptr));
-    } else if (isa<linalg::VecmatOp>(op)) {
+    } else if (isa<linalg::VecmatOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::VecmatOp *>(nullptr));
-    } else if (isa<linalg::BatchMatmulOp>(op)) {
+    } else if (isa<linalg::BatchMatmulOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchMatmulOp *>(nullptr));
-    } else if (isa<linalg::BatchMatvecOp>(op)) {
+    } else if (isa<linalg::BatchMatvecOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchMatvecOp *>(nullptr));
-    } else if (isa<linalg::BatchVecmatOp>(op)) {
+    } else if (isa<linalg::BatchVecmatOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchVecmatOp *>(nullptr));
-    } else if (isa<linalg::MatmulTransposeAOp>(op)) {
+    } else if (isa<linalg::MatmulTransposeAOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::MatmulTransposeAOp *>(nullptr));
-    } else if (isa<linalg::MatmulTransposeBOp>(op)) {
+    } else if (isa<linalg::MatmulTransposeBOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::MatmulTransposeBOp *>(nullptr));
-    } else if (isa<linalg::BatchMatmulTransposeAOp>(op)) {
+    } else if (isa<linalg::BatchMatmulTransposeAOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchMatmulTransposeAOp *>(nullptr));
-    } else if (isa<linalg::BatchMatmulTransposeBOp>(op)) {
+    } else if (isa<linalg::BatchMatmulTransposeBOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchMatmulTransposeBOp *>(nullptr));
+    } else if (isa<linalg::Conv2DNchwFchwOp>(linalgOp)) {
+      replaceOpInputs(static_cast<linalg::Conv2DNchwFchwOp *>(nullptr));
```

Comment:
Why not also handle other linalg conv* named ops similarly. This way, there won't be a surprising performance discrepancy between them. No need for additional tests, or just minimal ones.

Reply:
I have added all the conv2d cases and the grouped convolution ones.

Reply:
I'll add the tests.